src/playbackstrategy/msestrategy.js

```diff
@@ -548,6 +554,21 @@ function MSEStrategy(mediaSources, windowType, mediaKind, playbackElement, isUHD
     mediaPlayer.on(DashJSEvents.GAP_JUMP, onGapJump)
     mediaPlayer.on(DashJSEvents.GAP_JUMP_TO_END, onGapJump)
     mediaPlayer.on(DashJSEvents.QUOTA_EXCEEDED, onQuotaExceeded)
+    mediaPlayer.on(DashJSEvents.TEXT_TRACKS_ADDED, disableTextTracks)
+  }
+
+  function disableTextTracks() {
```

Comment:
What calls this function? Should `setSubtitles` call this when `state === false`?

Reply:
Hi @eirikbjornr thanks for taking a look. The thought behind this is that calling `disableTextTracks` when a `TECT_TRACK` is added is a failsafe way of preventing Dash.js from ever showing subtitles before BSP has signalled a need for them. As a result, I don't believe it's required on `setSubtitles(false)`, if it's helpful to add it in for consistency, I'm happy to do so.

Reply:
Ah I didn't spot it was added as a handler on `TEXT_TRACKS_ADDED` – that all makes sense now